src/schemes.js

```diff
@@ -82,9 +82,14 @@ add('yellowgreen',       'YlGn');
 add('yelloworangebrown', 'YlOrBr');
 add('yelloworangered',   'YlOrRd');
 
-export default function(name, scheme) {
+export default function(name, scheme, discreteScheme) {
   if (arguments.length > 1) {
     schemes[name] = scheme;
+
+    if (discreteScheme) {
+      discrete[name] = discreteScheme;
```

Comment:
Thanks! A discrete scheme is an array of arrays, indexed by count. As a result, while this assignment works fine for looking up a discrete scheme with a supported color count, the assignment to the `schemes` registry on line 87 is not valid. For all discrete schemes, we want to have a "backup" interpolator that can be used for any cases not explicitly covered by the explicit discrete color arrays. 

So, when adding a discrete scheme, we should also do something like the following:

```js
schemes[name] = interpolateRgbBasis(peek(scheme));
```

This takes the last discrete entry (the one with the most colors) and uses it create the backup interpolator (here created by using basis splines in the RGB color space).

Reply:
Mmh I understand. The idea was that one would pass this interpolation function as the 2nd parameter `scheme` and the discrete scheme as the 3rd.
This would result in the same structure as for the built in discrete schemes.

But I see that this is not really obvious. We could at least add a check if the 2nd parameter is a function if the 3rd parameter is a discrete scheme (and also test the structure of that) and throw an error if this is not the case.

What do you think?